ksql-common/src/test/java/io/confluent/ksql/function/UdfFactoryTest.java

```diff
@@ -48,6 +57,37 @@ public void shouldThrowIfNoVariantFoundThatAcceptsSuppliedParamTypes() {
     factory.getFunction(ImmutableList.of(Schema.Type.STRING, Schema.Type.INT64));
   }
 
+  @Test
+  public void shouldFindFirstMatchingFunctionWhenNullTypeInArgs() {
+    final KsqlFunction expected = new KsqlFunction(Schema.STRING_SCHEMA,
+        Collections.singletonList(Schema.STRING_SCHEMA),
+        functionName,
+        TestFunc.class
+    );
+    factory.addFunction(expected);
+    factory.addFunction(new KsqlFunction(Schema.STRING_SCHEMA,
+        Collections.singletonList(Schema.INT64_SCHEMA),
+        functionName,
+        TestFunc.class
+    ));
+
+    final KsqlFunction function = factory.getFunction(Collections.singletonList(null));
+    assertThat(function, equalTo(expected));
+  }
+
+  @Test
+  public void shouldNotMatchingFunctionWhenNullTypeInArgsIfParamLengthsDiffer() {
```

Comment:
Would be good to add a test case which checks that an exception is raised when there is a null parameter, with a matching number of arguments, but none of the non null argument types match.